source/particles/particle.cc

```diff
@@ -180,6 +180,14 @@ namespace Particles
       property_pool->deallocate_properties_array(properties);
   }
 
+  template <int dim, int spacedim>
+  void
+  Particle<dim, spacedim>::free_properties()
+  {
+    if (property_pool != nullptr && properties != PropertyPool::invalid_handle)
+      property_pool->deallocate_properties_array(properties);
+  }
+
```

Comment:
I am not that familiar with the particle infrastructure: why is this needed? What is the effect?

Reply:
So this took me totally offguard. When you overwrite the particle to which an iterator points to, the destructor of that particle is not called. Consequently, it still has memory allocated in the memory pool and when the destructor of the memory pool is called, it crashes (because it asserts that the particles have de-allocated it's memory). Consequently, I had to add this routine to correctly clean the memory associated with particles before "overwriting them". Took me a while to figure this out. In the next PR I would replace this with a function that overwrites the data in the same memory space. This would prevent needless allocation/deallocation.